core/camel-core-languages/src/main/java/org/apache/camel/language/csimple/CSimpleHelper.java

```diff
@@ -163,6 +178,40 @@ public static String bodyOneLine(Exchange exchange) {
         return body;
     }
 
+    public static String prettyBody(Exchange exchange) {
+        String body = exchange.getIn().getBody(String.class);
+
+        if (body == null) {
+            return null;
+        } else if (body.startsWith("{") && body.endsWith("}") || body.startsWith("[") && body.endsWith("]")) {
+            body = Jsoner.prettyPrint(body); //json
+        } else if (body.startsWith("<") && body.endsWith(">")) {
+            return CSimpleHelper.prettyXml(body); //xml
+        }
+
+        return body;
+    }
+
+    private static String prettyXml(String rawXml) {
+        try {
+            InputSource src = new InputSource(new StringReader(rawXml));
```

Comment:
Please, can you confirm this won't leak? I am under the impression `InputSource` won't close it (same thing if you use a `InputStream`).

Reply:
I will replace by `org.apache.camel.util.xml.XmlPrettyPrinter`